### PR TITLE
Enabled xml doc generation

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public interface IServer : IDisposable
         /// <summary>
         /// Start the server with an HttpApplication.
         /// </summary>
-        /// <param name="application">An instance of <see cref="IHttpApplication"/>.</param>
+        /// <param name="application">An instance of <see cref="IHttpApplication{TContext}"/>.</param>
         void Start<TContext>(IHttpApplication<TContext> application);
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Hmm, is it a bug that we don't have to require the type param ref?

**Comment by NTaylorMullen** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Talked offline

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs [hunk 2]
```diff
@@ -19,7 +19,8 @@ public interface IServer : IDisposable
         /// <summary>
         /// Start the server with an HttpApplication.
         /// </summary>
-        /// <param name="application">An instance of <see cref="IHttpApplication"/>.</param>
+        /// <param name="application">An instance of <see cref="IHttpApplication{TContext}"/>.</param>
+        /// <typeparam name="TContext">The context associated with the HttpApplication.</typeparam>
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Not sure where `HttpApplication` came from - is that a type or something? Should it just say `application`?

Comment (Eilon) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Or is it `IHttpApplication<T>`, in which case link it?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: I just used the same wording we have already used https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IHttpApplication.cs#L11 and https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs#L20.
But there is no `HttpApplication` as such.

Comment (Eilon) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Yuck. I would just say "application" then (lowercase). Can you patch up those two comments as well? I'm guessing they're copied from Katana or something.

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Nope, this all has to do with the gross generics.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs]: Done. 19501c03aa5d1973d7e5237b66af063dc21bb902

**Pull request merged**